150627359 Fix Accounts Tests

rnjane opened the pull request:
#### What does this PR do?
Fix Accounts Tests.
#### Description of Task to be completed?
- Adds asserts to existing incomplete accounts tests.
- Adds new tests, considered appropriate.
#### How should this be manually tested?
- run the application tests
```
python manage.py test
```
- All 153 tests should be passing.
#### What are the relevant pivotal tracker stories?
The Accounts tests need fixing -> #150627359

@Elbertbiggs360 @margierain

Commit:
feat(Accounts): Fix Account Tests

- Assert login redirects accordingly
- Assert failed login redirects accordingly

[Feature #150627359]

Commit:
feat(Accounts): Fix Accounts tests

- Assert that a user was created
- Assert that check is associated with the new user

[Feature #150627359]

Commit:
feat(Accounts): Fix Accounts tests

- Test for switching teams
- Test for checking team membership
- Test switching to own a team

[Feature #150627359]

Commit:
feat(Accounts): Fix Accounts tests

- Test a profile is created through the middle ware.

[Feature #150627359]

Commit:
feat(Accounts): Fix Account Tests

- Test creating an api key.
- Test revoking an api key.
- Test sending the correct report email.
- Test sending the correct set password email.

[Feature #150627359]

Commit:
feat(Accounts): Additional Accounts tests

- Test invalid credentials can't login.
- Test valid credentials redirects to checks page
- Test new user is redirected to setting up account
- Test login page is rendered
- Test login with wrong username and token redirects accordingly.

[Delivers #150627359]

margierain commented on hc/accounts/tests/test_check_token.py:
```diff
@@ -22,7 +23,18 @@ def test_it_redirects(self):
     ### Login and test it redirects already logged in
+    def test_login_redirects(self):
+        self.client.post("/accounts/check_token/alice/secret-token/")
+        r = self.client.get("/accounts/check_token/alice/secret-token/")
```
use a better variable name than r

margierain commented on hc/accounts/tests/test_check_token.py:
```diff
@@ -22,7 +23,18 @@ def test_it_redirects(self):
+        r = self.client.get("/accounts/check_token/alice/secret-token/")
+        self.assertRedirects(r, "/checks/")
 
     ### Login with a bad token and check that it redirects
```
remove the comments describing the tests

Commit:
feat(Accounts): Fix Accounts Tests

- Improve on comments in the application.
- Better variable naming.

[Feature #150627359]

NgangaMaryanne commented on hc/accounts/tests/test_login.py:
```diff
@@ -16,22 +17,55 @@ def test_it_sends_link(self):
-        r = self.client.post("/accounts/login/", form)
-        assert r.status_code == 302
+        response = self.client.post("/accounts/login/", form)
+        assert response.status_code == 302
```
other tests use the format `self.assert`. Its good practice to one format of code, to avoid confusion. Please change this to follow the `self.assert` format.

NgangaMaryanne commented on hc/accounts/tests/test_profile.py:
```diff
@@ -1,90 +1,109 @@
-        r = self.client.post("/accounts/profile/", form)
-        assert r.status_code == 200
+        response = self.client.post("/accounts/profile/", form)
+        assert response.status_code == 200
```
See comment above

NgangaMaryanne commented on hc/accounts/tests/test_profile.py:
```diff
@@ -1,90 +1,109 @@
-        r = self.client.post("/accounts/profile/", form)
-        assert r.status_code == 403
+        response = self.client.post("/accounts/profile/", form)
+        assert response.status_code == 403
```
⬆️

NgangaMaryanne commented on hc/accounts/tests/test_profile.py:
```diff
@@ -1,90 +1,109 @@
-        r = self.client.post("/accounts/profile/", form)
-        assert r.status_code == 200
+        response = self.client.post("/accounts/profile/", form)
+        assert response.status_code == 200
```
⬆️

NgangaMaryanne commented on hc/accounts/tests/test_profile.py:
```diff
@@ -1,90 +1,109 @@
-        r = self.client.post("/accounts/profile/", form)
-        assert r.status_code == 403
+        response = self.client.post("/accounts/profile/", form)
+        assert response.status_code == 403
```
⬆️

Commit:
feat(accounts):Fix accounts tests

- Change to self.assert format in test assertions

[Delivers #150627359]

NgangaMaryanne commented on hc/accounts/tests/test_login.py:
```diff
@@ -16,22 +17,55 @@ def test_it_sends_link(self):
+        removes it """
         self.client.session["bad_link"] = True
         self.client.get("/accounts/login/")
         assert "bad_link" not in self.client.session
```
Assert format used is `self.assert` this does not follow convention. Please change it.

NgangaMaryanne commented on hc/accounts/tests/test_profile.py:
```diff
@@ -1,90 +1,109 @@
-        r = self.client.post("/accounts/profile/", form)
-        assert r.status_code == 200
+        response = self.client.post("/accounts/profile/", form)
+        assert response.status_code == 200
```
see comment above

Commit:
feat(accounts):Fix Accounts Tests

- Refactor test assertions to use self.assert format

[Delivers #150627359]

NgangaMaryanne commented:
LGTM
